pkg/kudoctl/util/repo/repo.go

```diff
@@ -1,15 +1,123 @@
 package repo
 
-import "github.com/kudobuilder/kudo/pkg/apis/kudo/v1alpha1"
+import (
+	"fmt"
+	"os"
+
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1alpha1"
+	"github.com/spf13/afero"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	// Version is the repo / packaging version
+	Version         = "v1"
+	defaultRepoName = "community"
+)
 
 // RepositoryConfiguration represents a collection of parameters for operator repository.
 type RepositoryConfiguration struct {
-	URL string `json:"url"`
+	URL  string `json:"url"`
+	Name string `json:"name"`
+}
+
+// Repositories represents the repositories.yaml file usually in the $KUDO_HOME
+type Repositories struct {
+	RepoVersion  string                     `json:"repoVersion"`
+	Context      string                     `json:"context"`
+	Repositories []*RepositoryConfiguration `json:"repositories"`
 }
 
 // Default initialized repository.
 var Default = &RepositoryConfiguration{
-	URL: "https://kudo-repository.storage.googleapis.com",
+	Name: defaultRepoName,
+	URL:  "https://kudo-repository.storage.googleapis.com",
+}
+
+// NewRepoFile creates a new repo with only defaults populated
+func NewRepoFile() *Repositories {
+	return &Repositories{
+		RepoVersion:  Version,
+		Context:      defaultRepoName,
+		Repositories: []*RepositoryConfiguration{Default},
+	}
+}
+
+// Add appends a slice of repo configs to repositories file
+func (r Repositories) Add(repo ...*RepositoryConfiguration) {
+	r.Repositories = append(r.Repositories, repo...)
+}
+
+// SetContext switches the context to another repo config in the repositories file.  errors if no repo found.
+func (r Repositories) SetContext(context string) error {
```

Comment:
we're not using this right now. Could we push this to another PR then? adding code that is not used in PR is not a good practice...